assets/js/blocks-registry/register-inner-block.js

```diff
@@ -4,29 +4,36 @@
 import { registeredBlocks } from './registered-blocks-init';
 
 /**
- * Registers an inner block that can be added as a child of another block.
+ * Validates that an option is of the given type. Otherwise, throws an error.
  *
- * @export
- * @param {string}   main      Name of the parent block.
- * @param {string}   value     Name of the child block being registered.
- * @param {function} component React component used to render the child block.
+ * @throws Will throw an error if the type of the option doesn't match the expected type.
+ * @param {Object} options      Object containing the option to validate.
+ * @param {string} optionName   Name of the option to validate.
+ * @param {string} expectedType Type expected for the option.
  */
-export function registerInnerBlock( { main, blockName, component } ) {
-	if ( typeof main !== 'string' ) {
-		throw new Error(
-			'Wrong parent name when registering an inner block. `main` must be a string.'
-		);
-	}
-	if ( typeof blockName !== 'string' ) {
-		throw new Error(
-			'Wrong block name when registering an inner block. `blockName` must be a string.'
-		);
-	}
-	if ( typeof component !== 'function' ) {
+const validateOption = ( options, optionName, expectedType ) => {
+	if ( typeof options[ optionName ] !== expectedType ) {
 		throw new Error(
-			'Wrong component when registering an inner block. `component` must be a function.'
+			`Incorrect value for the ${ optionName } argument when registering an inner block. It must be a ${ expectedType }.`
 		);
 	}
+};
```

Comment:
I think I'd prefer `assertOption` here because it throws an error.  The term "validate" is less assertive and I've noticed a pattern where prefixing a function with `validate` returns bool.  I'll make the change.